lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2019/ashrae_90_1_2019.AirLoopHVAC.rb

```diff
@@ -453,4 +432,173 @@ def air_loop_hvac_energy_recovery_ventilator_flow_limit(air_loop_hvac, climate_z
+        max_oa_frac_sch_type = 'Schedule:Constant'
+      elsif max_oa_frac_sch.to_ScheduleCompact.is_initialized
+        max_oa_frac_sch = max_oa_frac_sch.to_ScheduleCompact.get
+        max_oa_frac_sch_type = 'Schedule:Compact'
+      end
+    end
+
+    # Add EMS to "cap" the OA calculated by the
+    # Controller:MechanicalVentilation object
+    # to the design v_ot using the maximum OA
+    # fraction schedule
+
+    # Add EMS sensors
+    # OA mass flow calculated by the Controller:MechanicalVentilation
+    air_loop_hvac_name_ems = "EMS_#{air_loop_hvac.name.to_s.gsub(' ', '_')}"
+    oa_vrp_mass_flow = OpenStudio::Model::EnergyManagementSystemSensor.new(air_loop_hvac.model, 'Air System Outdoor Air Mechanical Ventilation Requested Mass Flow Rate')
+    oa_vrp_mass_flow.setKeyName(air_loop_hvac.name.to_s)
+    oa_vrp_mass_flow.setName("#{air_loop_hvac_name_ems}_OA_VRP")
+    # Actual sensed OA mass flow
+    oa_mass_flow = OpenStudio::Model::EnergyManagementSystemSensor.new(air_loop_hvac.model, 'Air System Outdoor Air Mass Flow Rate')
+    oa_mass_flow.setKeyName(air_loop_hvac.name.to_s)
+    oa_mass_flow.setName("#{air_loop_hvac_name_ems}_OA")
+    # Actual sensed volumetric OA flow
+    oa_vol_flow = OpenStudio::Model::EnergyManagementSystemSensor.new(air_loop_hvac.model, 'System Node Standard Density Volume Flow Rate')
+    oa_vol_flow.setKeyName("#{air_loop_hvac.name} Mixed Air Node")
+    oa_vol_flow.setName("#{air_loop_hvac_name_ems}_SUPPLY_FLOW")
+
+    # Add EMS actuator
+    max_oa_fraction = OpenStudio::Model::EnergyManagementSystemActuator.new(max_oa_frac_sch, max_oa_frac_sch_type, 'Schedule Value')
+    max_oa_fraction.setName("#{air_loop_hvac_name_ems}_MAX_OA_FRAC")
+
+    # Add EMS program
+    max_oa_ems_prog = OpenStudio::Model::EnergyManagementSystemProgram.new(air_loop_hvac.model)
+    max_oa_ems_prog.setName("#{air_loop_hvac.name}_MAX_OA_FRAC")
+    max_oa_ems_prog_body = <<-EMS
+    IF #{air_loop_hvac_name_ems}_OA > #{air_loop_hvac_name_ems}_OA_VRP,
+    SET #{air_loop_hvac_name_ems}_MAX_OA_FRAC = NULL,
+    ELSE,
+    IF #{air_loop_hvac_name_ems}_SUPPLY_FLOW > 0,
+    SET #{air_loop_hvac_name_ems}_MAX_OA_FRAC = #{v_ot} / #{air_loop_hvac_name_ems}_SUPPLY_FLOW,
+    ELSE,
+    SET #{air_loop_hvac_name_ems}_MAX_OA_FRAC = NULL,
+    ENDIF,
+    ENDIF
+    EMS
+    max_oa_ems_prog.setBody(max_oa_ems_prog_body)
+
+    max_oa_ems_prog_manager = OpenStudio::Model::EnergyManagementSystemProgramCallingManager.new(air_loop_hvac.model)
+    max_oa_ems_prog_manager.setName("SET_#{air_loop_hvac.name.to_s.gsub(' ', '_')}_MAX_OA_FRAC")
+    max_oa_ems_prog_manager.setCallingPoint('InsideHVACSystemIterationLoop')
+    max_oa_ems_prog_manager.addProgram(max_oa_ems_prog)
+
+    # Hard-size the sizing:system
+    # object with the calculated min OA flow rate
+    sizing_system = air_loop_hvac.sizingSystem
+    sizing_system.setDesignOutdoorAirFlowRate(v_ot)
+    sizing_system.setSystemOutdoorAirMethod('ZoneSum')
+
+    return true
+  end
```

Comment:
New method that sets the MDP based on `v_oz` and `v_ot` based on a simplified `e_v` calculation.